pkg/postage/batchstore/reserve_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package batchstore_test
+
+import (
+	"io/ioutil"
+	"math/big"
+	"math/rand"
+	"os"
+	"testing"
+
+	"github.com/ethersphere/bee/pkg/postage"
+	"github.com/ethersphere/bee/pkg/postage/batchstore"
+	postagetest "github.com/ethersphere/bee/pkg/postage/testing"
+	"github.com/ethersphere/bee/pkg/statestore/leveldb"
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+// TestBatchStoreUnreserve is testing the correct behaviour of the reserve.
+// the following assumptions are tested by on each modification of the batches (top up, depth increase, price change)
+// - reserve exceeds capacity
+// - value-consistency of unreserved POs
+func TestBatchStoreUnreserve(t *testing.T) {
+	// we cannot  use the mock statestore here since the iterator is not giving the right order
```

Comment:
can we somehow simplify this test and make it more readable by splitting assertions into test helpers that can be reused across different tests? also please consider table driven tests. it is very difficult to unpack this test

Reply:
will comment